app/src/main/java/com/duckduckgo/app/browser/autocomplete/BrowserAutoCompleteSuggestionsAdapter.kt

```diff
@@ -59,22 +65,21 @@ class BrowserAutoCompleteSuggestionsAdapter(
     }
 
     override fun getItemCount(): Int {
-        if (suggestions.isNotEmpty()) {
-            return suggestions.size
+        if (suggestions.isEmpty() && phrase.isNotBlank()) {
+            return 1 // No suggestions message
         }
-
-        // if there are no suggestions, we'll use a recycler row to display "no suggestions"
-        return 1
+        return suggestions.size
     }
 
     @UiThread
-    fun updateData(newSuggestions: List<AutoCompleteSuggestion>) {
-        suggestions.clear()
-        suggestions.addAll(newSuggestions)
+    fun updateData(newPhrase: String, newSuggestions: List<AutoCompleteSuggestion>) {
+        if (phrase == newPhrase && suggestions == newSuggestions) return
+        phrase = newPhrase
+        suggestions = newSuggestions
         notifyDataSetChanged()
     }
 
-    companion object {
+    object Type {
```

Comment:
should this be an object?

Reply:
Yes, this is intentional. These naturally live in the companion object but when there are multiple related items imho grouping them in their own object assists readability.